File: 2610. Convert an Array Into a 2D Array With Conditions/Solution.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int n = nums.length, high = 0;
        int[] counter = new int[n + 1];
        List<List<Integer>> res = new ArrayList<>();
        for (int i : nums) {
            counter[i]++;
            high = Math.max(high, counter[i]);
        }
        for (int i = 0; i < high; i++) {
            List<Integer> curr = new ArrayList<>();
            for (int j = 1; j <= n; j++) {
                if (counter[j] > 0) curr.add(j);
                counter[j]--;
            }
            res.add(curr);
        }
        return res;
    }
}

File: 2610. Convert an Array Into a 2D Array With Conditions/Solution.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        n, high, res = len(nums), 0, []
        counter = [0 for i in range(n + 1)]
        for i in nums:
            counter[i] += 1
            high = max(high, counter[i])
        for i in range(high):
            curr = []
            for j in range(1, n + 1):
                if counter[j] > 0: curr.append(j)
                counter[j] -= 1
            res.append(curr)
        return res